rfcs/0131-sri-hashes.md

```diff
@@ -0,0 +1,48 @@
+---
+feature: sri-hashes
+start-date: 2022-07-25
+author: Winter
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Use SRI hashes and the `hash` parameter for fetchers.
+
+# Motivation
+[motivation]: #motivation
+
+Currently in Nixpkgs, the hash formats that are used in parameters to fetchers range from SRI and base-32 (the most common ones) to hex. In addition to this, packages use the algorithm-specific arguments (`sha256`, `sha512`, etc.) or the generic `hash` argument at random. This creates inconsistencies that we'd rather not have.
+
+# Detailed design
+[design]: #detailed-design
+
+1. Require all new packages to use SRI hashes + `hash` when possible
+2. Do a treewide migration for the majority of cases (single fetcher in `src`, using a fetcher that supports `hash` and SRI hashes (`fetchFromGitHub`, `fetchurl`, etc.)) to `hash` + SRI hashes
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Some fetchers still don't support SRI hashes (i.e. [`fetchgit`](https://github.com/NixOS/nixpkgs/pull/79987), which is blocked due to [this discussion](https://github.com/NixOS/nixpkgs/pull/79987#discussion_r378735698) not being resolved)
```

Comment:
`fetchgit` actually supports it since https://github.com/NixOS/nixpkgs/pull/144693, there might be other fetchers that donʼt but I do not think it is an issue with any of the common ones. My problem with is that Eelco indicated `hash` is not a good fit and Nix builtins use different attribute name.

Reply:
I've removed the `fetchgit` drawback and have added the `hash` argument name to unresolved questions for further discussion.